Update how-datadog-agent-determines-the-hostname.md (#22114)
DataDog/documentation: content/en/agent/faq/how-datadog-agent-determines-the-hostname.md

If the name is recognized as a common non-unique name (for example, `localhost.localdomain`), the current rule fails and passes through to the next.

> See [Hostname detection logic](https://github.com/DataDog/datadog-agent/tree/main/pkg/util/hostname) for details on how the Agent determines the hostname.

Reviewer: ```suggestion
See [Hostname detection logic][3] for details on how the Agent determines the hostname.
```

> A more detailed explanation of how the Agent determines the hostname can be found on [our public repository on Github](https://github.com/DataDog/datadog-agent/tree/main/pkg/util/hostname).

Reviewer: Could you use the link list at the bottom of the doc to add the link?

```suggestion
See [Hostname detection Logic](https://github.com/DataDog/datadog-agent/tree/main/pkg/util/hostname) for details on how the Agent determines the hostname.
```
